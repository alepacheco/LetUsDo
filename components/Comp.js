function Home() {
  return <div>Imported</div>;
}

export default Home;