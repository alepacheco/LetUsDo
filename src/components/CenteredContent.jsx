import React from 'react';
import '../styles/components/CenteredContent.scss';

export const CenteredContent = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
