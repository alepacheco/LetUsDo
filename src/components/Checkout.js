import React from 'react';
import StripeCheckout from 'react-stripe-checkout';
import Button from 'react-bootstrap/Button';

export default class Checkout extends React.Component {
    constructor(props) {
        super(props);

        this.onToken = this.onToken.bind(this);
    }

    onToken(token) {
        fetch('/save-stripe-token', {
            method: 'POST',
            body: JSON.stringify(token),
        }).then(response => {
            response.json().then(data => {
                alert(`We are in business, ${data.email}`);
            });
        });
    }

    render() {
        return (
            <StripeCheckout
                name="We Do"
                description={this.props.description || 'Complete task'}
                panelLabel="Pay"
                amount={this.props.amount || 2000}
                currency="GBP"
                token={this.onToken}
                stripeKey="pk_test_TYooMQauvdEDq54NiTphI7jx"
            >
                <Button className="get-it-done-button">Get it done</Button>
            </StripeCheckout>
        )
    }
}