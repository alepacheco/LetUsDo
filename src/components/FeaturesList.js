import React from 'react';
import '../styles/components/FeaturesList.scss';
import cafe from '../static/images/cafe.png'
import london from '../static/images/london-bridge.png'
import cash from '../static/images/cash.png'

const ListElement = ({ image }) => {
  return (<div className="item">
    <img className="feature-icon" src={image}></img>
    <div className="list-element-title">Beautiful themes that are responsive and customizable</div>
    <div className="list-element-body">No design skills needed. You have complete control over the look and feel of your website, from its layout, to content</div>
  </div>);
};

export class FeaturesList extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="features-list">
        <div className="container">
          <ListElement image={london} />
          <ListElement image={cash} />
          <ListElement image={cafe} />
        </div>
      </div>
    );
  }
}

export default FeaturesList;
