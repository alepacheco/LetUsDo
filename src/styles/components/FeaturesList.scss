.features-list {
  .item {
    flex: 1;
    margin: 16px;
  }

  .list-element-title {
    font-weight: bold;
    font-size: 18px;
  }

  .list-element-body {
    font-size: 12px;
  }

  .feature-icon {
    height: 56px;
  }
}
