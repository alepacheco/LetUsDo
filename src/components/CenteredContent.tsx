import React from 'react';
import 'src/styles/components/CenteredContent.scss';

type CenteredContentProps = {
  reverse?: boolean;
};

export const CenteredContent: React.FC<CenteredContentProps> = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
