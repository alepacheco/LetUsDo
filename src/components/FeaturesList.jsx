import React from 'react';
import '../styles/components/FeaturesList.scss';
import cafe from '../static/images/cafe.png';
import london from '../static/images/london-bridge.png';
import cash from '../static/images/cash.png';
import CenteredContent from './CenteredContent';

const ListElement = ({ image, title, content }) => (
  <div className="item">
    <img className="feature-icon" src={image} />
    <div className="list-element-title">{title}</div>
    <div className="list-element-body">{content}</div>
  </div>
);

export const FeaturesList = props => {
  return (
    <div className="features-list">
      <CenteredContent>
        <ListElement image={london} title={'Based in London'} content={'ffff'} />
        <ListElement image={cash} title={'Fixed price for your task'} content={'ffff'} />
        <ListElement image={cafe} title={'Worries free service'} content={'ffff'} />
      </CenteredContent>
    </div>
  );
};

export default FeaturesList;
