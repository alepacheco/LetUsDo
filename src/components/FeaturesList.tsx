import React from 'react';
import 'src/styles/components/FeaturesList.scss';
// @ts-ignore
import cafe from 'src/static/images/cafe.png';
// @ts-ignore
import london from 'src/static/images/london-bridge.png';
// @ts-ignore
import cash from 'src/static/images/cash.png';
import CenteredContent from './CenteredContent';

type ListElementProps = {
  image?: string;
  title: string;
};


const ListElement: React.FC<ListElementProps> = ({ image, title, children }) => (
  <div className="item">
    <img alt="" className="feature-icon" src={image} />
    <div className="list-element-title">{title}</div>
    <div className="list-element-body">{children}</div>
  </div>
);


export const FeaturesList = () => (
  <div className="features-list">
    <CenteredContent>
      <ListElement image={london} title="Based in London">
        We are starting in London and have people on the streets ready to take on some work as soon
        as you send us a request.
      </ListElement>
      <ListElement image={cash} title="Flat fee">
        To make it even easier there is no pricing guide, you will always pay the same whatever the
        task at hand is. <b>Just 20Â£</b>
      </ListElement>
      <ListElement image={cafe} title="Worries free service">
        lorem ipsum dolor sit amet, consectetur adip lorem ipsum dolor sit amet, consectetur adip
        lorem ipsum dolor sit amet, consectetur adip lorem ipsum dolor sit amet
      </ListElement>
    </CenteredContent>
  </div>
);

export default FeaturesList;
