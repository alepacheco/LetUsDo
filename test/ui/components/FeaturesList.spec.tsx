/* eslint-disable no-undef, import/no-unresolved */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { FeaturesList } from 'components/FeaturesList';

describe('<FeaturesList />', () => {
  it('renders', () => {
    const wrapper = shallow(<FeaturesList />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
