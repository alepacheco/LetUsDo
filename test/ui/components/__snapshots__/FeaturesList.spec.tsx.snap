// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturesList /> renders 1`] = `
<div
  className="features-list"
>
  <Component>
    <ListElement
      image="static/images/london-bridge.png"
      title="Based in London"
    >
      We are starting in London and have people on the streets ready to take on some work as soon as you send us a request.
    </ListElement>
    <ListElement
      image="static/images/cash.png"
      title="Flat fee"
    >
      To make it even easier there is no pricing guide, you will always pay the same whatever the task at hand is. 
      <b>
        Just 20Â£
      </b>
    </ListElement>
    <ListElement
      image="static/images/cafe.png"
      title="Worries free service"
    >
      lorem ipsum dolor sit amet, consectetur adip lorem ipsum dolor sit amet, consectetur adip lorem ipsum dolor sit amet, consectetur adip lorem ipsum dolor sit amet
    </ListElement>
  </Component>
</div>
`;
